kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/main/view/EcosystemActivity.java

```diff
@@ -0,0 +1,186 @@
+	public static final String ECOSYSTEM_ORDER_HISTORY_FRAGMENT_TAG = "ecosystem_order_history_fragment_tag";
+
+	private IBalancePresenter balancePresenter;
+	private IEcosystemPresenter ecosystemPresenter;
+	private IMarketplacePresenter marketplacePresenter;
+
+	@Override
+	protected int getLayoutRes() {
+		return R.layout.kinecosystem_activity_main;
+	}
+
+	@Override
+	protected int getTitleRes() {
+		return R.string.kinecosystem_kin_marketplace;
+	}
+
+	@Override
+	protected int getNavigationIcon() {
+		return R.drawable.kinecosystem_ic_back;
+	}
+
+	@Override
+	protected View.OnClickListener getNavigationClickListener() {
+		return new View.OnClickListener() {
+			@Override
+			public void onClick(View v) {
+				onBackPressed();
+			}
+		};
+	}
+
+	@Override
+	protected void onCreate(@Nullable Bundle savedInstanceState) {
+		super.onCreate(savedInstanceState);
+		IBalanceView balanceView = findViewById(R.id.balance_view);
+		balancePresenter = new BalancePresenter(balanceView, EventLoggerImpl.getInstance(),
+			BlockchainSourceImpl.getInstance(), OrderRepository.getInstance());
+		balancePresenter.setClickListener(new BalanceClickListener() {
+			@Override
+			public void onClick() {
+				ecosystemPresenter.balanceItemClicked();
+			}
+		});
+		ecosystemPresenter = new EcosystemPresenter(this, this);
+	}
+
+	@Override
+	public void updateTitle(String title) {
+		getToolbar().setTitle(title);
+	}
+
+	@Override
+	public void navigateToMarketplace() {
+		MarketplaceFragment marketplaceFragment = (MarketplaceFragment) getSupportFragmentManager()
+			.findFragmentByTag(ECOSYSTEM_MARKETPLACE_FRAGMENT_TAG);
+		if (marketplaceFragment == null) {
+			marketplaceFragment = MarketplaceFragment.newInstance();
+		}
+
+		marketplacePresenter = getMarketplacePresenter(marketplaceFragment);
+
+		getSupportFragmentManager().beginTransaction()
+			.replace(R.id.fragment_frame, marketplaceFragment, ECOSYSTEM_MARKETPLACE_FRAGMENT_TAG)
+			.commit();
+
+		setVisibleScreen(MARKETPLACE);
+
+	}
+
+	private IMarketplacePresenter getMarketplacePresenter(MarketplaceFragment marketplaceFragment) {
+		if (marketplacePresenter == null) {
+			marketplacePresenter = new MarketplacePresenter(marketplaceFragment, OfferRepository.getInstance(),
+				OrderRepository.getInstance(),
+				BlockchainSourceImpl.getInstance(),
+				this,
+				EventLoggerImpl.getInstance());
+		}
+
+		return marketplacePresenter;
+	}
+
+	private void setVisibleScreen(@ScreenId final int id) {
+		ecosystemPresenter.visibleScreen(id);
+		balancePresenter.visibleScreen(id);
+	}
+
+	@Override
+	public void navigateToOrderHistory(boolean isFirstSpendOrder) {
+		OrderHistoryFragment orderHistoryFragment = (OrderHistoryFragment) getSupportFragmentManager()
+			.findFragmentByTag(ECOSYSTEM_ORDER_HISTORY_FRAGMENT_TAG);
+
+		if (orderHistoryFragment == null) {
+			orderHistoryFragment = OrderHistoryFragment.newInstance();
+		}
+
+		new OrderHistoryPresenter(orderHistoryFragment,
```

Comment:
I'm afraid that if OrderHistoryFragment will get destroyed and recreated by the OS it will not include the OrderHistoryPresenter anymore. Can we simply the flow and the fragment will create the presenter and then attached to it in onCreateview ?